hail/src/main/scala/is/hail/types/physical/stypes/concrete/SNDArraySlice.scala

```diff
@@ -82,6 +82,30 @@ class SNDArraySliceValue(
     new SNDArraySliceValue(st, otherShape, strides, firstDataAddress)
   }
 
+  // FIXME: only optimized for column major
+  override def setToZero(cb: EmitCodeBuilder): Unit = {
+    val eltType = pt.elementType.asInstanceOf[PNumeric with PPrimitive]
+
+    val contiguousDims = contiguousDimensions(cb)
+
+    def recur(startPtr: Value[Long], dim: Int): Unit =
+      if (dim > 0) {
+        cb.ifx(contiguousDims.ceq(dim), {
```

Comment:
It seems like making this a check inside each loop is wrong -- we'll evaluate this check many times. Can we evaluate continuousDims per dim from 0 to nDims until we get a False, and put those in an array or something, then look that up?

I'm fine with this being optimized for column-major, but it's *really* bad for not-column-major.

Reply:
Good catch, though I don't see how building an array of bools helps, you still have the branches in the inner loops.

Only option I see is to lift the branches to the top level at the cost of a bit of code growth. See what you think.